File: progress.py
import time

def printProgressBar (iteration, total, prefix = '', suffix = '', decimals = 1, length = 100, fill = '█'):
    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))

    # Number of fill chacters required
    filledLength = int(length * iteration // total)
    bar = (fill * filledLength) + ('-' * (length - filledLength))
    print(f'\r{prefix} |{bar}| {percent}% {suffix}', end = '\r')

for i in range(100):
    printProgressBar(i, 100, "Downloading", "Goal", 0, 100, "*")
    time.sleep(.1)

File: progress.js
const sleep = require('system-sleep');
const e = require('ecoji-js');

function printProgressBar(iteration, total, prefix, suffix, fill) {

    // Write out the prefix
    process.stdout.write(prefix + " | ");

    // Current progress using the fill character
    for (let i = 0; i < iteration; i++) {
        process.stdout.write(fill);
    }

    // Remaining progress using the "-" character
    for (let i = iteration; i < total; i++) {
        process.stdout.write('-');
    }

    // Write out the suffix
    process.stdout.write(" | " + suffix);
    process.stdout.write('\r');
}

s = e.encode("3").substring(0, 2);
for (let i = 0; i < 100; i++) {
    printProgressBar(i, 100, "Downloading", "TICKTOCK", s);
    sleep(25);
}
